src/main/java/commands/RemoveCommand.java

```diff
@@ -23,13 +36,69 @@ public RemoveCommand(int removeIndex) {
     }
 
     @Override
-    public void execute(CardList cards, Ui ui, Access access, Storage storage) {
+    public void execute(CardList cards, Ui ui, Access access, Storage storage)
+            throws IncorrectAccessLevelException, IOException {
+        if (access.isAdminLevel()) {
+            removeModule(ui, access, storage);
+        } else if (access.isModuleLevel()) {
+            removeChapter(ui, access, storage);
+        } else if (access.isChapterLevel()) {
+            removeCard(ui, access, storage);
+        } else {
+            throw new IncorrectAccessLevelException("Remove command can only be called at admin, "
+                    + "module and chapter level.\n");
+        }
+    }
+
+    private void removeModule(Ui ui, Access access, Storage storage) throws IOException {
+        try {
+            ModuleList modules = access.getAdmin().getModules();
+            ArrayList<Module> allModules = modules.getAllModules();
+            Module module = allModules.get(removeIndex);
+            File directory = new File(storage.getFilePath() + "/" + module.toString());
+            boolean isRemoved = storage.deleteDirectory(directory);
+            if (!isRemoved) {
+                ui.showError("There was a problem deleting module in directory.");
```

Comment:
Perhaps can extract this part as an exception so that someone reading the code wouldn't get distracted by alternative paths taken when error conditions happen.